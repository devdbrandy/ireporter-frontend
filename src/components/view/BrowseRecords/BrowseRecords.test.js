import React from 'react';
import { shallow } from 'enzyme';
import Modal from 'react-modal';
import { BrowseRecords } from './index';

jest.mock('react-modal');

/**
 * Wrapper for enzyme shallow component
 *
 * @returns {object} The mocked props and component
 */
const setup = () => {
  const props = {
    records: [],
    fetchRecords: jest.fn(),
  };

  const component = shallow(<BrowseRecords {...props} />);
  return { props, component };
};

describe('<BrowseRecords />', () => {
  it('renders without crashing given the required props', () => {
    Modal.setAppElement.mockImplementation(() => null);

    const { component } = setup();
    expect(component).toMatchSnapshot();
  });
});
