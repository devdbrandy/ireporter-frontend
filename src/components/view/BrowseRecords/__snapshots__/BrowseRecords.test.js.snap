// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BrowseRecords /> renders without crashing given the required props 1`] = `
<Fragment>
  <Layout>
    <withRouter(Panel)
      title="Browse Records"
    >
      <div
        className="records-list"
      >
        <div
          className="row"
        >
          <Record />
        </div>
      </div>
    </withRouter(Panel)>
  </Layout>
</Fragment>
`;
